api/src/pcapi/core/providers/allocine_movie_list.py

```diff
@@ -24,10 +26,14 @@ def _upsert_product(products_by_allocine_id: dict[int, Product], movie: Allocine
     product = products_by_allocine_id.get(allocine_id)
     movie_data = _build_movie_data(movie)
     if not product:
+        allocine_products_provider = Provider.query.filter(
```

Comment:
Pour éviter de faire cet appel autant de fois qu'il y a de produit tu pourrais faire la requête dans `synchronize_products` et le passer en argument à _upsert_product

Reply:
Of course